include/ghex/structured/simple_field_wrapper.hpp

```diff
@@ -296,6 +296,9 @@ namespace structured {
 
         GT_FUNCTION
         void set_data(value_type* ptr) { m_data = ptr; }
+        
+        GT_FUNCTION
+        int num_components() const noexcept { return 1; }
```

Comment:
This goes with other comments: would it make sense to have a single wrapper to handle vector fields?
Also, should we choose between `wrapper` of `descriptor` consistently?